starter-code/database.js

```diff
@@ -101,43 +101,45 @@ class Database {
         // LAB 5
         // Implement the query to list all products
         // remeber once it's finish to comment callback('Error listing products');
-        
-        callback('Error listing products');
+        database.collection('product').find().toArray(callback);
+      //  callback('Error listing products');
       }
     });
   }
 
-  deleteProduct( productName, callback = (error, result) => {}) {
+  deleteProduct( name, callback = (error, result) => {}) {
     this.connect((error, database) => {
       if (error){
         callback(error);
       } else {
         // LAB 6
         // Implement the query to delete a product
-        // productName is the name of the producto to delete 
+        // productName is the name of the producto to delete
         // remeber once it's finish to comment callback('Error deleting product');
-        
-        callback('Error deleting product');
+        database.collection('product').deleteOne({name}, callback);
+        //callback('Error deleting product');
       }
     });
   }
 
   addProductToShoppingCart({userFirstName, productName}, callback = (error) => {}){
-    this.connect((error, database) => {
-      if (error) {
-        callback(error);
-      } else {
-        // LAB 7
-        // Implement the query to buy a product
-        // userFirstName is the name of user who purchase the product
-        // productName is the name of the product that we want to buy
-        // Think if you may need to implement two queries chained
-        // remeber once it's finish to comment callback('Error buying product');
-        
-        callback('Error buying product');
+        this.connect((error, database) => {
+          if (error) {
+            callback(error);
+          } else {
+            // LAB 7
+            // Implement the query to buy a product
+            // userFirstName is the name of user who purchase the product
+            // productName is the name of the product that we want to buy
+            // Think if you may need to implement two queries chained
+            // remeber once it's finish to comment callback('Error buying product');
+            database.collection('product').find({"name" : productName}).toArray((error, product) => {
```

Comment:
for the next time look if there is method that does all that bunch of code, because there is, check for findOneAndUpdate()